arch/openrisc/lib/memset.S

```diff
@@ -0,0 +1,95 @@
+/*
+ * OpenRISC memset.S
+ *
+ * Hand-optimized assembler version of memset for OpenRISC.
+ * Algorithm inspired by several other arch-specific memset routines
+ * in the kernel tree
+ *
+ * Copyright (C) 2015 Olof Kindgren <olof.kindgren@gmail.com>
+ *
+ *      This program is free software; you can redistribute it and/or
+ *      modify it under the terms of the GNU General Public License
+ *      as published by the Free Software Foundation; either version
+ *      2 of the License, or (at your option) any later version.
+ */
+
+	.global memset
+	.type	memset, @function
+memset:
+	#Args r3 = *s, r4 = c, r5 = n
+	#r13, r15, r17, r19 used as temp regs
+
+	#Exit if n == 0
+	l.sfeqi		r5, 0
+	l.bf		4f
+
+	#Truncate c to char
+	l.andi  	r13, r4, 0xff
+
+	#Skip word extension if c is 0
+	l.sfeqi		r13, 0
+	l.bf		1f
+	#Check for at least two whole words (8 bytes) in delay slot
```

Comment:
This comment sounds like you are looking in the delay slot to find two words.
Instead of commenting on that an instruction is in a delay-slot, the coding style (invented by Jonas?),
that we use to denote delay slot is to indent the instruction in the delay slot by one extra space.
I.e.
```
l.bf            1f
 l.sfleui       r5, 7
```
This style is used extensively in the openrisc asm code in the kernel,
as well in other projects as well.